llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -4033,14 +4033,29 @@ MachineInstr *SIInstrInfo::convertToThreeAddress(MachineInstr &MI,
   if (Src0Literal && !ST.hasVOP3Literal())
     return nullptr;
 
-  unsigned NewOpc = IsFMA ? IsF16 ? AMDGPU::V_FMA_F16_gfx9_e64
-                                  : IsF64 ? AMDGPU::V_FMA_F64_e64
-                                          : IsLegacy
-                                                ? AMDGPU::V_FMA_LEGACY_F32_e64
-                                                : AMDGPU::V_FMA_F32_e64
-                          : IsF16 ? AMDGPU::V_MAD_F16_e64
-                                  : IsLegacy ? AMDGPU::V_MAD_LEGACY_F32_e64
-                                             : AMDGPU::V_MAD_F32_e64;
+  auto getNewFMAInst = [&]() {
```

Comment:
```suggestion
  auto getNewFMAInst = [=]() {
```

I would just move this to a full fledged helper function instead of a lambda. It would also be more straightforward to just switch over all the possibilities, rather than going backwards from these IsFMA /  IsF16 / IsLegacy derived from lists of opcodes above

Reply:
done